Test suite fails 16 tests on hppa

### What did you do?
Built and run Pillow master (290cd170eccd52b285cfbe379b5915a2d46281ee) on Gentoo/HPPA (big endian!) and ran the test suite:

### What actually happened?
Tests/test_file_gif.py::TestFileGif::test_extents FAILED 
Tests/test_file_ico.py::TestFileIco::test_unexpected_size FAILED
Tests/test_file_jpeg.py::TestFileJpeg::test_photoshop FAILED
Tests/test_file_libtiff.py::TestFileLibTiff::test_16bit_RGB_tiff FAILED
Tests/test_file_libtiff.py::TestFileLibTiff::test_old_style_jpeg FAILED
Tests/test_file_libtiff.py::TestFileLibTiff::test_strip_cmyk_16l_jpeg FAILED
Tests/test_file_libtiff.py::TestFileLibTiff::test_strip_ycbcr_jpeg_1x1_sampling FAILED
Tests/test_file_libtiff.py::TestFileLibTiff::test_strip_ycbcr_jpeg_2x2_sampling FAILED
Tests/test_file_libtiff.py::TestFileLibTiff::test_tiled_ycbcr_jpeg_1x1_sampling FAILED
Tests/test_file_libtiff.py::TestFileLibTiff::test_tiled_ycbcr_jpeg_2x2_sampling FAILED
Tests/test_file_png.py::TestFilePng::test_sanity FAILED
Tests/test_image_getextrema.py::TestImageGetExtrema::test_extrema FAILED
Tests/test_image_getextrema.py::TestImageGetExtrema::test_true_16 FAILED
Tests/test_imagedraw.py::TestImageDraw::test_arc_width_pieslice_large FAILED
Tests/test_imagedraw.py::TestImageDraw::test_ellipse_width_large FAILED
Tests/test_imagedraw.py::TestImageDraw::test_rectangle_I16 FAILED

Not all of those are regressions, here is the fail list for 6.0.0:

Tests/test_file_libtiff.py::TestFileLibTiff::test_old_style_jpeg FAILED
Tests/test_file_libtiff.py::TestFileLibTiff::test_strip_ycbcr_jpeg_1x1_sampling FAILED
Tests/test_file_libtiff.py::TestFileLibTiff::test_strip_ycbcr_jpeg_2x2_sampling FAILED
Tests/test_file_libtiff.py::TestFileLibTiff::test_tiled_ycbcr_jpeg_1x1_sampling FAILED
Tests/test_file_libtiff.py::TestFileLibTiff::test_tiled_ycbcr_jpeg_2x2_sampling FAILED
Tests/test_file_png.py::TestFilePng::test_sanity FAILED
Tests/test_image_getextrema.py::TestImageGetExtrema::test_extrema FAILED
Tests/test_image_getextrema.py::TestImageGetExtrema::test_true_16 FAILED

### What are your OS, Python and Pillow versions?

* OS: Gentoo HPPA
* Python: 2.7.15
* Pillow: current master

[pillow-6.1.0.log](https://github.com/python-pillow/Pillow/files/3343316/pillow-6.1.0.log)

Just ran the testsuite for 9.9.0 on ppc32 which now says:
```
[...]
XFAIL Tests/test_decompression_bomb.py::TestDecompressionBomb::test_exception_ico
  different exception
XFAIL Tests/test_file_webp_animated.py::test_write_animation_L
  Fails on big-endian
XFAIL Tests/test_file_webp_animated.py::test_write_animation_RGB
  Fails on big-endian
XFAIL Tests/test_image_resample.py::TestCoreResampleAlphaCorrect::test_levels_rgba
  Current implementation isn't precise enough
XFAIL Tests/test_image_resample.py::TestCoreResampleAlphaCorrect::test_levels_la
  Current implementation isn't precise enough
XPASS Tests/test_file_palm.py::test_p_mode Palm P image is wrong
FAILED Tests/test_image.py::TestImage::test_zero_tobytes[size0] - SystemError: tile cannot extend out...
FAILED Tests/test_image.py::TestImage::test_zero_tobytes[size1] - SystemError: tile cannot extend out...
FAILED Tests/test_image.py::TestImage::test_zero_tobytes[size2] - SystemError: tile cannot extend out...
FAILED Tests/test_imagepath.py::test_invalid_coords - SystemError: new style getargs format but argum...
====== 4 failed, 2653 passed, 223 skipped, 5 xfailed, 1 xpassed, 5 warnings in 172.01s (0:02:52) =======
```
So my originally reported failures for ppc32 are fixed or the tests get skipped where appropriate.

Current failures are:
```
__________________________________ TestImage.test_zero_tobytes[size0] __________________________________

self = <Tests.test_image.TestImage object at 0xf0d37730>, size = (1, 0)

    @pytest.mark.parametrize("size", ((1, 0), (0, 1), (0, 0)))
    def test_zero_tobytes(self, size):
        im = Image.new("RGB", size)
>       assert im.tobytes() == b""

Tests/test_image.py:792: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <PIL.Image.Image image mode=RGB size=1x0 at 0xF0D37C40>, encoder_name = 'raw', args = 'RGB'
e = <ImagingEncoder object at 0xf043ba40>

    def tobytes(self, encoder_name="raw", *args):
        """
        Return image as a bytes object.
    
        .. warning::
    
            This method returns the raw image data from the internal
            storage.  For compressed image data (e.g. PNG, JPEG) use
            :meth:`~.save`, with a BytesIO parameter for in-memory
            data.
    
        :param encoder_name: What encoder to use.  The default is to
                             use the standard "raw" encoder.
        :param args: Extra arguments to the encoder.
        :returns: A :py:class:`bytes` object.
        """
    
        # may pass tuple instead of argument list
        if len(args) == 1 and isinstance(args[0], tuple):
            args = args[0]
    
        if encoder_name == "raw" and args == ():
            args = self.mode
    
        self.load()
    
        # unpack data
        e = _getencoder(self.mode, encoder_name, args)
>       e.setimage(self.im)
E       SystemError: tile cannot extend outside image

../.local/lib/python3.10/site-packages/PIL/Image.py:722: SystemError
__________________________________ TestImage.test_zero_tobytes[size1] __________________________________

self = <Tests.test_image.TestImage object at 0xf107fd90>, size = (0, 1)

    @pytest.mark.parametrize("size", ((1, 0), (0, 1), (0, 0)))
    def test_zero_tobytes(self, size):
        im = Image.new("RGB", size)
>       assert im.tobytes() == b""

Tests/test_image.py:792: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <PIL.Image.Image image mode=RGB size=0x1 at 0xF107F520>, encoder_name = 'raw', args = 'RGB'
e = <ImagingEncoder object at 0xf043b7a0>

    def tobytes(self, encoder_name="raw", *args):
        """
        Return image as a bytes object.
    
        .. warning::
    
            This method returns the raw image data from the internal
            storage.  For compressed image data (e.g. PNG, JPEG) use
            :meth:`~.save`, with a BytesIO parameter for in-memory
            data.
    
        :param encoder_name: What encoder to use.  The default is to
                             use the standard "raw" encoder.
        :param args: Extra arguments to the encoder.
        :returns: A :py:class:`bytes` object.
        """
    
        # may pass tuple instead of argument list
        if len(args) == 1 and isinstance(args[0], tuple):
            args = args[0]
    
        if encoder_name == "raw" and args == ():
            args = self.mode
    
        self.load()
    
        # unpack data
        e = _getencoder(self.mode, encoder_name, args)
>       e.setimage(self.im)
E       SystemError: tile cannot extend outside image

../.local/lib/python3.10/site-packages/PIL/Image.py:722: SystemError
__________________________________ TestImage.test_zero_tobytes[size2] __________________________________

self = <Tests.test_image.TestImage object at 0xf107f310>, size = (0, 0)

    @pytest.mark.parametrize("size", ((1, 0), (0, 1), (0, 0)))
    def test_zero_tobytes(self, size):
        im = Image.new("RGB", size)
>       assert im.tobytes() == b""

Tests/test_image.py:792: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <PIL.Image.Image image mode=RGB size=0x0 at 0xF107F940>, encoder_name = 'raw', args = 'RGB'
e = <ImagingEncoder object at 0xf043bb00>

    def tobytes(self, encoder_name="raw", *args):
        """
        Return image as a bytes object.
    
        .. warning::
    
            This method returns the raw image data from the internal
            storage.  For compressed image data (e.g. PNG, JPEG) use
            :meth:`~.save`, with a BytesIO parameter for in-memory
            data.
    
        :param encoder_name: What encoder to use.  The default is to
                             use the standard "raw" encoder.
        :param args: Extra arguments to the encoder.
        :returns: A :py:class:`bytes` object.
        """
    
        # may pass tuple instead of argument list
        if len(args) == 1 and isinstance(args[0], tuple):
            args = args[0]
    
        if encoder_name == "raw" and args == ():
            args = self.mode
    
        self.load()
    
        # unpack data
        e = _getencoder(self.mode, encoder_name, args)
>       e.setimage(self.im)
E       SystemError: tile cannot extend outside image

../.local/lib/python3.10/site-packages/PIL/Image.py:722: SystemError
_________________________________________ test_invalid_coords __________________________________________

    def test_invalid_coords():
        # Arrange
        coords = ["a", "b"]
    
        # Act / Assert
        with pytest.raises(ValueError) as e:
>           ImagePath.Path(coords)
E           SystemError: new style getargs format but argument is not a tuple

Tests/test_imagepath.py:74: SystemError
```

--- a/Tests/test_file_webp_animated.py
+++ b/Tests/test_file_webp_animated.py
@@ -1,6 +1,7 @@
 import pytest
+from packaging.version import parse as parse_version
 
-from PIL import Image
+from PIL import Image, features
 
 from .helper import (
     assert_image_equal,
@@ -27,7 +28,6 @@ def test_n_frames():
         assert im.is_animated
 
 
-@pytest.mark.xfail(is_big_endian(), reason="Fails on big-endian")
 def test_write_animation_L(tmp_path):
     """
     Convert an animated GIF to animated WebP, then compare the frame count, and first
@@ -46,6 +46,11 @@ def test_write_animation_L(tmp_path):
             orig.load()
             im.load()
             assert_image_similar(im, orig.convert("RGBA"), 32.9)
+
+            if is_big_endian():
+                webp = parse_version(features.version_module("webp"))
+                if webp < parse_version("1.2.2"):
+                    return
             orig.seek(orig.n_frames - 1)
             im.seek(im.n_frames - 1)
             orig.load()
@@ -53,7 +58,6 @@ def test_write_animation_L(tmp_path):
             assert_image_similar(im, orig.convert("RGBA"), 32.9)
 
 
-@pytest.mark.xfail(is_big_endian(), reason="Fails on big-endian")
 def test_write_animation_RGB(tmp_path):
     """
     Write an animated WebP from RGB frames, and ensure the frames
@@ -69,6 +73,10 @@ def test_write_animation_RGB(tmp_path):
             assert_image_equal(im, frame1.convert("RGBA"))
 
             # Compare second frame to original
+            if is_big_endian():
+                webp = parse_version(features.version_module("webp"))
+                if webp < parse_version("1.2.2"):
+                    return
             im.seek(1)
             im.load()
             assert_image_equal(im, frame2.convert("RGBA"))

--- a/Tests/test_file_webp_animated.py
+++ b/Tests/test_file_webp_animated.py
@@ -50,7 +50,7 @@ def test_write_animation_L(tmp_path):
             if is_big_endian():
                 webp = parse_version(features.version_module("webp"))
                 if webp < parse_version("1.2.2"):
-                    return
+                    pytest.skip("Fails with libwebp earlier than 1.2.2")
             orig.seek(orig.n_frames - 1)
             im.seek(im.n_frames - 1)
             orig.load()
@@ -76,7 +76,7 @@ def test_write_animation_RGB(tmp_path):
             if is_big_endian():
                 webp = parse_version(features.version_module("webp"))
                 if webp < parse_version("1.2.2"):
-                    return
+                    pytest.skip("Fails with libwebp earlier than 1.2.2")
             im.seek(1)
             im.load()
             assert_image_equal(im, frame2.convert("RGBA"))

